pipenv/project.py

```diff
@@ -388,9 +389,22 @@ def settings(self):
         """A dictionary of the settings added to the Pipfile."""
         return self.parsed_pipfile.get('pipenv', {})
 
-    @property
-    def scripts(self):
-        return dict(self.parsed_pipfile.get('scripts', {}))
+    def has_script(self, name):
+        try:
+            return name in self.parsed_pipfile['scripts']
+        except KeyError:
+            return False
+
+    def build_script(self, name, extra_args=None):
+        try:
+            script = Script.parse(self.parsed_pipfile['scripts'][name])
+        except KeyError:
+            script = Script([name])
+        except ValueError:
+            raise ValueError('invalid script entry {0!r}'.format(name))
```

Comment:
The only way you can get here is like this, right?

```
[scripts]
tests = "bash ./run-tests.sh"
blah = ""
```

This ends up looking kinda gross to the user:

```
Loading .env environment variables…
Traceback (most recent call last):
  File "/Users/jtratner/pipenv/pipenv/project.py", line 400, in build_script
    script = Script.parse(self.parsed_pipfile['scripts'][name])
  File "/Users/jtratner/pipenv/pipenv/cmdparse.py", line 22, in parse
    return cls(value)
  File "/Users/jtratner/pipenv/pipenv/cmdparse.py", line 14, in __init__
    raise ValueError('invalid script')
ValueError: invalid script

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/jtratner/.pyenv/versions/3.6.4/bin/pipenv", line 11, in <module>
    load_entry_point('pipenv', 'console_scripts', 'pipenv')()
  File "/Users/jtratner/pipenv/pipenv/vendor/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/Users/jtratner/pipenv/pipenv/vendor/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/Users/jtratner/pipenv/pipenv/vendor/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/jtratner/pipenv/pipenv/vendor/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/jtratner/pipenv/pipenv/vendor/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/Users/jtratner/pipenv/pipenv/cli.py", line 599, in run
    core.do_run(command=command, args=args, three=three, python=python)
  File "/Users/jtratner/pipenv/pipenv/core.py", line 2252, in do_run
    do_run_posix(command, args)
  File "/Users/jtratner/pipenv/pipenv/core.py", line 2214, in do_run_posix
    script = project.build_script(command, args)
  File "/Users/jtratner/pipenv/pipenv/project.py", line 404, in build_script
    raise ValueError('invalid script entry {0!r}'.format(name))
ValueError: invalid script entry 'blah'
```

Usability-wise, perhaps a better feedback would look like:

```
$ pipenv run blah
Can't run *blah* - it's an empty script.
$ echo $?
1
```

I'm not 100% clear the extra indirection is worth it here, but I understand where you're coming from :)